### ISSUE TITLE
Address missing tandems

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/my-solar-system#281]
Running Studio with `&phetioPrintMissingTandems` currently reports that 420 required tandems are missing, and 2509 optional tandems are missing.  I'm having a lot of trouble understanding the console output, to identify where I need to add tandems.  I'll need synchronous help from either @samreid or @zepumph.  Also assigning to @AgustinVallejo to ride along, so he can see how this is done.

**Comment by pixelzoom** [comment]
Almost half of the missing required tandems were due to an instrumentation bug in KeypadDialog, see https://github.com/phetsims/scenery-phet/commit/ae1de72efaefd0cdbd77163d0bb346c82c8046bb. So now we're down to 232

**Comment by pixelzoom** [comment]
Down to 82 missing required tandems.

The way I'm identifying these is to just guess based on the tandemName shown in the console.  The phetioID (e.g. "mySolarSystem.requiredTandem.thumbNode") is more misleading than helpful.  And the stack traces are difficult to navigate and (in most cases) not helpful.

**Comment by pixelzoom** [comment]
Down to 46 missing required tandems.  Based on the tandem names, I'm guessing that they are related to Slider and some sort of button.

**Comment by zepumph** [comment]
> The way I'm identifying these is to just guess based on the tandemName shown in the console. The phetioID (e.g. "mySolarSystem.requiredTandem.thumbNode") is more misleading than helpful. And the stack traces are difficult to navigate and (in most cases) not helpful.

Yeah, the tandem is quite gross, and the stack trace is manual and often hard to parse. I don't know a better way to mark these entities. We could add another mode that could debugger for each required tandem that isn't passed so you can handle the hard ones one at a time. Do you have any other ideas or work flows that you wish you could do?

**Comment by pixelzoom** [comment]
32 error messages were due to a single HSlider, now `Tandem.OPT_OUT` in the above commit.

Down to 14 missing required tandems.

**Comment by pixelzoom** [comment]
I finally located the needle in the haystack.  This stack trace pointed me to `enterButton: RectangularPushButton`  in KeypadDialog: 
```
"Error
    at Tandem.onMissingTandem (http://localhost:8080/chipper/dist/js/tandem/js/Tandem.js:121:26)
    at Emitter.initializePhetioObject (http://localhost:8080/chipper/dist/js/tandem/js/PhetioObject.js:136:38)
    at new PhetioObject (http://localhost:8080/chipper/dist/js/tandem/js/PhetioObject.js:122:12)
    at new PhetioDataHandler (http://localhost:8080/chipper/dist/js/tandem/js/PhetioDataHandler.js:73:5)
    at new Emitter (http://localhost:8080/chipper/dist/js/axon/js/Emitter.js:34:5)
    at new PushButtonModel (http://localhost:8080/chipper/dist/js/sun/js/buttons/PushButtonModel.js:40:25)
    at new RectangularPushButton (http://localhost:8080/chipper/dist/js/sun/js/buttons/RectangularPushButton.js:38:29)
    at new KeypadDialog (http://localhost:8080/chipper/dist/js/scenery-phet/js/keypad/KeypadDialog.js:107:25)
    at new ValuesPanel (http://localhost:8080/chipper/dist/js/my-solar-system/js/common/view/ValuesPanel.js:49:26)
    at new MySolarSystemScreenView (http://localhost:8080/chipper/dist/js/my-solar-system/js/common/view/MySolarSystemScreenView.js:114:24)"
```

Instrumented in the above commit.  We now have 0 missing required tandems, and can look at removing `phet-io.validation: false` from package.json.

**Comment by pixelzoom** [comment]
I finally located the needle in the haystack.  This stack trace pointed me to `enterButton: RectangularPushButton`  in KeypadDialog.  The other 13 stack traces were related to this, and were of no help.
```
"Error
    at Tandem.onMissingTandem (http://localhost:8080/chipper/dist/js/tandem/js/Tandem.js:121:26)
    at Emitter.initializePhetioObject (http://localhost:8080/chipper/dist/js/tandem/js/PhetioObject.js:136:38)
    at new PhetioObject (http://localhost:8080/chipper/dist/js/tandem/js/PhetioObject.js:122:12)
    at new PhetioDataHandler (http://localhost:8080/chipper/dist/js/tandem/js/PhetioDataHandler.js:73:5)
    at new Emitter (http://localhost:8080/chipper/dist/js/axon/js/Emitter.js:34:5)
    at new PushButtonModel (http://localhost:8080/chipper/dist/js/sun/js/buttons/PushButtonModel.js:40:25)
    at new RectangularPushButton (http://localhost:8080/chipper/dist/js/sun/js/buttons/RectangularPushButton.js:38:29)
    at new KeypadDialog (http://localhost:8080/chipper/dist/js/scenery-phet/js/keypad/KeypadDialog.js:107:25)
    at new ValuesPanel (http://localhost:8080/chipper/dist/js/my-solar-system/js/common/view/ValuesPanel.js:49:26)
    at new MySolarSystemScreenView (http://localhost:8080/chipper/dist/js/my-solar-system/js/common/view/MySolarSystemScreenView.js:114:24)"
```

`keypadDialog.enterButton` was instrumented in the above commit.  We now have 0 missing required tandems, and can look at removing `phet-io.validation: false` from package.json.

**Comment by pixelzoom** [comment]
The sim is showing no errors in Studio with `phetioValidation=true`, so it seems safe to remove `phet-io.validation: false` from package.json.

**Comment by pixelzoom** [comment]
`phet-io.validation: false` has been removed from package.json.  So validation will now be done by Studio and CT.  I'll leave this open until I've verified that there are no CT problems.

**Comment by pixelzoom** [comment]
CT is happy. Closing.
